File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/utils/HTMLUtil.java
package ch.rmy.android.http_shortcuts.utils;

import android.content.Context;
import android.os.Build;
import android.support.annotation.StringRes;
import android.text.Html;
import android.text.Spanned;

public class HTMLUtil {

    public static Spanned getHTML(Context context, @StringRes int stringRes) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            return Html.fromHtml(context.getString(stringRes), 0);
        } else {
            return Html.fromHtml(context.getString(stringRes));
        }
    }

}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/utils/HTMLUtil.kt
package ch.rmy.android.http_shortcuts.utils

import android.text.Html
import android.text.Spanned

object HTMLUtil {
    fun toSpanned(string: String): Spanned =
        fromHTML(string.convertNewlines().normalize())

    private fun String.normalize(): String =
        replace("<pre>", "<tt>")
            .replace("</pre>", "</tt>")

    private fun String.convertNewlines() =
        removeSuffix("\n").replace("\n", "<br>")

    private fun fromHTML(string: String): Spanned =
        Html.fromHtml(string, 0, null, null)
}
